src/caches/sdirk_caches.jl

```diff
@@ -50,6 +56,13 @@ DiffEqBase.@def iipnlcachefields begin
     jac_config = nothing
     linsolve = nothing
     z₊ = similar(z)
+    if typeof(z) <: AbstractArray
```

Comment:
for iip it has to be <:AbstractArray, so this branch isn't doing anything. Just keep the top